### PR TITLE
[Mailer] Add support for the profiler

### LINKED ISSUE #31592: Mailer does not collect profiling
I was not sure if opening this issue, since Mailer component is experimental and 4.3 version is still in beta. Anyway, may is about to end and it looks like an important feature is missing in Mailer. I'm not even sure if this is a bug or a new feature (I opted for last one, being a new component)

My expectation is to replace current combo of Swiftmailer library and Swifmailer bundle with new combo of Mailer and MIME components, and to get more or less the same functionalities.
So, for example, I expect to see some info in the profiler and in the web debug toolbar (you know, the old good envelope icon linked to info about headers, text, rendered message, etc.).

I couldn't find anything in current mailer, for example the NullTransport is doing absolutely nothing

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | closes #31592
| License       | MIT
| Doc PR        | n/a

Web profiler for the Mailer.

### CODE REVIEW

**Comment by fabpot** [comment]
image

**Review by ro0NL — COMMENTED** [review]
can we port the iframe approach from swiftmailer for rendering HTML while at it?

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/mailer.html.twig [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            padding: .3em .5em;
+        }
+    </style>
+{% endblock %}
+
+{% block menu %}
+    <span class="label {{ collector.messages|length ? '' : 'disabled' }}">
+        <span class="icon">{{ include('@WebProfiler/Icon/mailer.svg') }}</span>
+
+        <strong>Emails</strong>
```
[the remark this answers is not in the document]
Comment (fabpot) [reply src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/mailer.html.twig]: haha, I've done the change from `E-mail` (Swiftmailer bundle) to `Email`. I think Email is better and use by everyone.

[the remark this answers is not in the document]
Comment (ro0NL) [reply src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/mailer.html.twig]: i wasnt sure indeed compared to swiftmailer :) but both are correct, `email` is the common case :+1:

**Comment by fabpot** [comment]
I think it's finished now. It supports multiple transports and I've added a visual representation of the message parts.

image

Regarding the iframe for HTML, it's not simple and I'd like to do it in a follow-up PR (or maybe someone else can work on it :)).

**Comment by fabpot** [comment]
Travis failure will be fixed when 4.4 is merged to master.

**Review by javiereguiluz — APPROVED** [review]
Nice! I've reviewed this and I can't see any obvious error. It's true that we're missing the "HTML preview" feature, but as Fabien said, that can be done in separate PRs after merging this one.

**Comment by lyrixx** [comment]
> Regarding the iframe for HTML, it's not simple and I'd like to do it in a follow-up PR (or maybe someone else can work on it :)).

did anyone worked on this ?

I did that on a project to embed `phpinfo()` in a debug page (because we don't trust our hosting platform :joy: and we needed to check things)

```php
            <iframe
                src="data:text/html;base64;charset=utf-8,{{ phpinfo|raw|base64_encode }}"
                style="height: 80vh;width: 100%;"
            >
            </iframe>
```

(Note: the `raw` looks useless...)

**Comment by ro0NL** [comment]
> did anyone worked on this ?

Yes, see swiftmailerbundle :)

**Comment by lyrixx** [comment]
@ro0NL But it's not in symfony/mailer, right ?

**Pull request merged**